File: staticfiles/js/settings/update_password.js
document.getElementById('passwordForm').addEventListener('submit', async function(event) {
    event.preventDefault(); // Предотвращаем стандартную отправку формы

    const form = this;
    const formData = new FormData(form);
    const csrftoken = document.querySelector('[name=csrfmiddlewaretoken]').value;

    // Очищаем предыдущие уведомления
    let alertDiv = document.querySelector('.alert');
    if (alertDiv) alertDiv.remove();

    try {
        // Отправляем данные на фронтенд для логирования
        const frontendLogResponse = await fetch('/settings/change-password/', {
            method: 'POST',
            body: formData,
            headers: {
                'X-CSRFToken': csrftoken
            }
        });
        const frontendLogData = await frontendLogResponse.json();
        console.log('Ответ от фронтенда:', frontendLogData);

        // Отправляем основной запрос на бэкенд
        const backendResponse = await fetch(form.action, {
            method: form.method,
            body: formData,
            headers: {
                'X-CSRFToken': csrftoken
            }
        });
        const backendData = await backendResponse.json();

        // Создаём уведомление
        const alertDiv = document.createElement('div');
        alertDiv.className = `alert alert-${backendData.status === 'success' ? 'success' : 'danger'} mt-3`;
        alertDiv.textContent = backendData.message;
        form.parentNode.insertBefore(alertDiv, form);

        if (backendData.status === 'success') {
            // Очищаем поля формы
            form.reset();
            // Обновляем страницу через 2 секунды
            setTimeout(() => {
                window.location.reload();
            }, 3000);
        }
    } catch (error) {
        console.error('Ошибка AJAX:', error);
        const alertDiv = document.createElement('div');
        alertDiv.className = 'alert alert-danger mt-3';
        alertDiv.textContent = 'Произошла ошибка при отправке запроса.';
        form.parentNode.insertBefore(alertDiv, form);
    }
});

// Переключение видимости пароля
function togglePasswordVisibility(inputId, toggleId) {
    const input = document.getElementById(inputId);
    const toggle = document.getElementById(toggleId);

    toggle.addEventListener('click', function() {
        const isPassword = input.type === 'password';
        input.type = isPassword ? 'text' : 'password';
        toggle.classList.toggle('bi-eye', !isPassword);
        toggle.classList.toggle('bi-eye-slash', isPassword);
    });
}

// Применяем переключение для всех полей
togglePasswordVisibility('currentPassword', 'toggleCurrentPassword');
togglePasswordVisibility('newPassword', 'toggleNewPassword');
togglePasswordVisibility('confirmPassword', 'toggleConfirmPassword');

File: apps/settings/update_password.py
from django.views.decorators.http import require_POST
from django.http import JsonResponse
import re


@require_POST
def update_password(request):
    """Обрабатывает AJAX-запрос для смены пароля."""
    try:
        # Получаем данные из формы
        current_password = request.POST.get('current_password')
        new_password = request.POST.get('new_password')
        confirm_password = request.POST.get('confirm_password')

        # Заглушка: случайный текущий пароль (для примера)
        placeholder_password = 'hardcode'
        print(f"Заглушка текущего пароля: {placeholder_password}")

        # Проверка текущего пароля
        if current_password != placeholder_password:
            print("Ошибка: Неверный текущий пароль")
            return JsonResponse({
                "status": "error",
                "message": "Неверный текущий пароль"
            }, status=400)

        # Проверка совпадения нового пароля и подтверждения
        if new_password != confirm_password:
            print("Ошибка: Новый пароль и подтверждение не совпадают")
            return JsonResponse({
                "status": "error",
                "message": "Новый пароль и подтверждение не совпадают"
            }, status=400)

        # Проверка требований к новому паролю
        if len(new_password) < 8:
            print("Ошибка: Пароль должен содержать минимум 8 символов")
            return JsonResponse({
                "status": "error",
                "message": "Пароль должен содержать минимум 8 символов"
            }, status=400)
        if not re.search(r'[A-Z]', new_password):
            print("Ошибка: Пароль должен содержать хотя бы одну заглавную букву")
            return JsonResponse({
                "status": "error",
                "message": "Пароль должен содержать хотя бы одну заглавную букву"
            }, status=400)
        if not re.search(r'\d', new_password):
            print("Ошибка: Пароль должен содержать хотя бы одну цифру")
            return JsonResponse({
                "status": "error",
                "message": "Пароль должен содержать хотя бы одну цифру"
            }, status=400)
        if not re.search(r'[!@#$%^&*(),.?":{}|<>]', new_password):
            print("Ошибка: Пароль должен содержать хотя бы один специальный символ")
            return JsonResponse({
                "status": "error",
                "message": "Пароль должен содержать хотя бы один специальный символ"
            }, status=400)

        # Если все проверки пройдены
        print("Пароль успешно изменён")
        return JsonResponse({
            "status": "success",
            "message": "Пароль успешно изменён"
        })

    except Exception as e:
        print(f"Ошибка: {str(e)}")
        return JsonResponse({
            "status": "error",
            "message": f"Произошла ошибка: {str(e)}"
        }, status=500)